tests/test_deserialize_sequence.cc

```diff
@@ -22,13 +22,30 @@
 
 // SPDX-License-Identifier: LGPL-2.1-or-later
 
+#include <filesystem>
+#include <fstream>
+
 #include <gul14/catch.h>
 
 #include "deserialize_sequence.h"
+#include "internals.h"
 
 using namespace task;
 using namespace std::literals;
 
+namespace {
+
+static const std::filesystem::path temp_dir{ "unit_test_files" };
+
+void remove_temp_dir()
+{
+    if (std::filesystem::exists(temp_dir))
+        std::filesystem::remove_all(temp_dir);
+    std::filesystem::create_directory(temp_dir);
+}
```

Comment:
We already remove the folder `unit_test_files` in the `main()` function of the test executable:
```cpp

#include <filesystem>

#define CATCH_CONFIG_RUNNER
#include <gul14/catch.h>

int main(int argc, char* argv[])
{
    if (std::filesystem::exists("unit_test_files"))
        std::filesystem::remove_all("unit_test_files");

    return Catch::Session().run(argc, argv);
}
```
If we remove it again in some test, we'll lose some of the output of other tests. If removing something is really necessary, it should be a subfolder.

Reply:
Oh ha, I didn't think about that now. I removed the `remove_all` function.

Anyhow to solve the problem of a duplicated temporary folder name I move it to a header in the test folder and reference it where it is needed. The header file is called `details_unit_test.h`:

```c++
#include <filesystem>

#define DETAILS_UNIT_TEST

namespace {

static const std::filesystem::path temp_dir{ "unit_test_files" };

} // namespace

#endif // DETAILS_UNIT_TEST
```

Is this a more suitable way to solve it?

Reply:
The idea is great, and now I remember that we stumbled over the exact same problem in libgit4cpp. The thing with the anonymous namespace means that every translation unit including the new header gets an independent copy of the `temp_dir` object. That is probably OK, but in libgit4cpp we simply used a regular function instead. Have a look at `tests/main.cc` and `tests/main.h`.